File: src/main/java/szewek/mcflux/util/InjectCond.java
package szewek.mcflux.util;

import szewek.fl.FLU;

public enum InjectCond {
	NONE, MOD, CLASS;

	public boolean check(String[] args) {
		switch(this) {
			case NONE:
				return true;
			case MOD:
				return args.length != 0 && FLU.loadedMods(args);
			case CLASS:
				for (String a : args) {
					try {
						if (Class.forName(a) != null)
							return true;
					} catch (ClassNotFoundException ignored) {
					}
				}
				break;
		}
		return false;
	}
}

File: src/main/java/szewek/mcflux/util/InjectCond.kt
package szewek.mcflux.util

import szewek.fl.FLU

enum class InjectCond {
	NONE, MOD, CLASS;

	fun check(args: Array<String>): Boolean {
		when (this) {
			NONE -> return true
			MOD -> return args.isNotEmpty() && FLU.loadedMods(*args)
			CLASS -> for (a in args) {
				try {
					if (Class.forName(a) != null)
						return true
				} catch (ignored: ClassNotFoundException) {
				}

			}
		}
		return false
	}
}
